zkmemory/src/ram_machine/memory.rs

```diff
@@ -152,70 +168,134 @@ where
     fn read(&mut self, address: K) -> CellInteraction<K, V> {
```

Comment:
Read should return a tuple `(V, CellInteraction<K, V>)`, we need to return the result that we read from the memory.